File: python/leetcode/base-7.py
class Solution:
    def convertToBase7(self, num):
        if num < 0:
            return '-' + self.convertToBase7(num * - 1)
        if num < 7:
            return str(num) + ''
        else:
            return str(self.convertToBase7(num // 7) + str(num % 7)) # Might need to call floor function or use //

File: java/leetcode/base-7.java
class Solution {
    public String convertToBase7(int num) {
        if (num < 0) {
            return "-" + convertToBase7(num * -1);
        }
        if (num < 7) {
            return String.valueOf(num);
        }
        return convertToBase7(Math.round(num / 7)) + num % 7;
    }
}
